samples/extensions/sparse_image/sparse_image.cpp

```diff
@@ -0,0 +1,1532 @@
+		get_memory_dependency_for_the_block(texture_block.column, texture_block.row, texture_block.old_mip_level, memory_index_list);
+
+		while (!memory_index_list.empty())
+		{
+			size_t memory_index = memory_index_list.front();
+			memory_index_list.pop_front();
+
+			if (!virtual_texture.page_table[memory_index].fixed)
+			{
+				virtual_texture.page_table[memory_index].render_required_list.remove({texture_block.old_mip_level, texture_block.column, texture_block.row});
+			}
+		}
+	}
+
+	if (!texture_block.on_screen)
+	{
+		return;
+	}
+
+	// New value calculations and placing into update and render_required lists
+	get_memory_dependency_for_the_block(texture_block.column, texture_block.row, texture_block.new_mip_level, memory_index_list);
+
+	while (!memory_index_list.empty())
+	{
+		size_t memory_index = memory_index_list.front();
+		memory_index_list.pop_front();
+
+		virtual_texture.page_table[memory_index].render_required_list.remove({texture_block.new_mip_level, texture_block.column, texture_block.row});
+		virtual_texture.page_table[memory_index].render_required_list.push_front({texture_block.new_mip_level, texture_block.column, texture_block.row});
+
+		if (!virtual_texture.page_table[memory_index].valid)
+		{
+			virtual_texture.update_list.remove(memory_index);
+			virtual_texture.update_list.push_back(memory_index);
+		}
+		else
+		{
+			continue;
+		}
+
+		std::list<MemPageDescription> mipgen_required_list;
+		struct MemPageDescription     mem_page_description = get_mem_page_description(memory_index);
+
+		mipgen_required_list.push_front(mem_page_description);
+
+		while (!mipgen_required_list.empty())
+		{
+			mem_page_description = mipgen_required_list.front();
+			mipgen_required_list.pop_front();
+
+			check_mip_page_requirements(mipgen_required_list, mem_page_description);
+		}
+	}
+}
+
+/**
+ * 	@brief Fill the MemPageDescription data structure.
+ */
+struct SparseImage::MemPageDescription SparseImage::get_mem_page_description(size_t memory_index)
+{
+	struct MemPageDescription mem_page_description = {};
+	uint8_t                   mip_level            = virtual_texture.mip_levels - 1U;
+
+	// TODO(GS): use get_mip_level() instead
+	while (memory_index < virtual_texture.mip_properties[mip_level].mip_base_page_index && mip_level > virtual_texture.base_mip_level)
+	{
+		mip_level--;
+	}
+
+	mem_page_description.mip_level = mip_level;
+	mem_page_description.x         = (memory_index - virtual_texture.mip_properties[mip_level].mip_base_page_index) % virtual_texture.mip_properties[mip_level].num_columns;
+	mem_page_description.y         = (memory_index - virtual_texture.mip_properties[mip_level].mip_base_page_index) / virtual_texture.mip_properties[mip_level].num_columns;
+
+	return mem_page_description;
+}
+
+/**
+ * 	@brief Get the memory_index of the particular page based on the MemPageDescription data structure.
+ */
+size_t SparseImage::get_memory_index(struct MemPageDescription mem_page_desc)
```

Comment:
Sometimes, you name it `memory_index`, sometimes you name it `page` or `page_index`. But it's always used as an index into the `page_table`.
Maybe a unified naming would be better.